includes/classes/EnqueueScript.php

```diff
@@ -0,0 +1,284 @@
+<?php
+/**
+ * This class use to register script.
+ *
+ * This class handles:
+ *   - Script dependencies.
+ *     This uses asset information to set script dependencies,
+ *     and version generated by @wordpress/dependency-extraction-webpack-plugin package.
+ *   - Script localization.
+ *     It also handles script translation registration.
+ *
+ * @package distributor
+ * @since   x.x.x
+ */
+
+namespace Distributor;
+
+use Exception;
+
+/**
+ * Class EnqueueScript
+ *
+ * @since x.x.x
+ */
+class EnqueueScript {
+	/**
+	 * Script ID.
+	 *
+	 * @since x.x.x
+	 */
+	private string $script_id;
+
+	/**
+	 * Script path relative to plugin root directory.
+	 *
+	 * @since x.x.x
+	 */
+	private string $relative_script_path;
+
+	/**
+	 * Script path absolute to plugin root directory.
+	 *
+	 * @since x.x.x
+	 */
+	private string $absolute_script_path;
+
+	/**
+	 * Script dependencies.
+	 *
+	 * @since x.x.x
+	 */
+	private array $script_dependencies = [];
+
+	/**
+	 * Script version.
+	 *
+	 * @since x.x.x
+	 */
+	private string $version = DT_VERSION;
+
+	/**
+	 * Flag to decide whether load script in footer.
+	 *
+	 * @since x.x.x
+	 */
+	private bool $load_script_in_footer = false;
+
+	/**
+	 * Flag to decide whether register script translation.
+	 *
+	 * @since x.x.x
+	 */
+	private bool $register_translations = false;
+
+	/**
+	 * Script localization parameter name.
+	 *
+	 * @since x.x.x
+	 */
+	private ?string $localize_script_param_name = null;
+
+	/**
+	 * Script localization parameter data.
+	 *
+	 * @since x.x.x
+	 */
+	private ?array $localize_script_param_data = null;
+
+	/**
+	 * Plugin root directory path.
+	 *
+	 * @since x.x.x
+	 */
+	private string $plugin_dir_path;
+
+	/**
+	 * Plugin root directory URL.
+	 *
+	 * @since x.x.x
+	 */
+	private string $plugin_dir_url;
+
+	/**
+	 * Plugin text domain.
+	 *
+	 * @since x.x.x
+	 */
+	private string $text_domain;
+
+	/**
+	 * EnqueueScript constructor.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $script_id   Script ID.
+	 * @param string $script_name Script name.
+	 *
+	 * @throws Exception If script file not found.
+	 */
+	public function __construct( string $script_id, string $script_name ) {
```

Comment:
```suggestion
	 * @param string $script_id   Script ID.
	 * @param string $script_file_name Script name.
	 *
	 * @throws Exception If script file not found.
	 */
	public function __construct( string $script_id, string $script_fle_name ) {
```

Took me a moment to figure out this wasn't the handle.